src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -293,34 +293,48 @@ internal List<WildcardPattern> MatchAll
             // Now search using the module path...
             foreach (string path in modulePath)
             {
-                // a name takes the form of .../moduleDir/moduleName.<ext>
-                // if only one name has been specified, then the moduleDir
-                // and moduleName are identical and repeated...
-                // Also append th ename if the path currently points at a directory
-                string qualifiedPath = Path.Combine(path, fileBaseName);
-
-                // Load the latest valid version if it is a multi-version module directory
-                module = LoadUsingMultiVersionModuleBase(qualifiedPath, manifestProcessingFlags, options, out found);
-
-                if (!found)
+#if UNIX
+                foreach (string folder in Directory.EnumerateDirectories(path))
                 {
-                    if (name.IndexOfAny(Utils.Separators.Directory) == -1)
+                    string fileName = Path.GetFileName(folder);
+                    if (String.Compare(fileName, name, StringComparison.OrdinalIgnoreCase) == 0)
                     {
-                        qualifiedPath = Path.Combine(qualifiedPath, fileBaseName);
+                        string qualifiedPath = folder;
+                        name = fileName;
+#else
+                        string qualifiedPath = Path.Combine(path, fileBaseName);
+#endif
+                        module = LoadUsingMultiVersionModuleBase(qualifiedPath, manifestProcessingFlags, options, out found);
+                        if (!found)
+                        {
+                            if (name.IndexOfAny(Utils.Separators.Directory) == -1)
+                            {
+#if UNIX
+                                qualifiedPath = Path.Combine(folder, fileName);
+#else
+                                qualifiedPath = Path.Combine(qualifiedPath, fileBaseName);
+#endif
```

Comment:
Perhaps we could exclude `#if` here and leave one line.

Reply:
I think this line isn't needed as I have qualifiedPath = folder above, was probably thinking this, but didn't put it together.  Will fix.